### PR TITLE
Font instructions and add eye icons

### PR DESCRIPTION
**Pull request by JamesLMilner**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

This PR provides instructions for adding new Icons to Dojo widgets. In doing so it updates the icon fonts to have the `eye` and `eye-slash` icons for future use in #792.

The PR also converts `Icon` to a functional widget.

Resolves #898 

Screenshot_20191118_203209

### CODE REVIEW

File: src/icon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	altText?: string;
+	/** Custom aria attributes */
+	aria?: { [key: string]: string | null };
+	/** Icon type, e.g. downIcon, searchIcon, etc. */
+	type: IconType;
+}
+
+const factory = create({ theme }).properties<IconProperties>();
+
+export const Icon = factory(function({ properties, middleware: { theme } }) {
```
Comment (tomdye) [inline src/icon/index.tsx]:
```suggestion
export const Icon = factory(function Icon({ properties, middleware: { theme } }) {
```

File: src/icon/index.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	const {
+		aria = {
+			hidden: 'true'
+		},
+		type,
+		altText
+	} = properties();
+
+	const themedCss = theme.classes(css);
```
Comment (tomdye) [inline src/icon/index.tsx]:
```suggestion
	const classes = theme.classes(css);
```

Comment (tomdye) [reply src/icon/index.tsx]: believe it's convention elsewhere to call this `classes`

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* dojo.widgets: https://dojowidgets-knw2280o8.now.sh

File: src/icon/index.tsx (hunk 2 above)
Comment (JamesLMilner) [reply src/icon/index.tsx]: :+1:

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* dojo.widgets: https://dojowidgets-eyxzn9mgk.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* dojo.widgets: https://dojowidgets-g9ilexq82.now.sh

**Pull request merged**